documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -1,20 +1,36 @@
 Pass Data to Algorithms
 #######################
 
-When using the C++ standard execution policies, oneDPL supports data being passed to its algorithms as specified
-in the ISO/IEC 14882:2017 standard (commonly called C++17). According to the standard, the calling code
-must prevent data races when using algorithms with parallel execution policies.
+When using the C++ standard aligned (or *host*) execution policies, oneDPL supports data being passed to its
+algorithms as specified in the ISO/IEC 14882:2017 standard (commonly called C++17). According to the standard,
+the calling code must prevent data races when using algorithms with parallel execution policies.
 
-Note: Implementations of ``std::vector<bool>`` are not required to avoid data races for concurrent modifications
-of vector elements. Some implementations may optimize multiple ``bool`` elements into a bitfield, making it unsafe
-for multithreading. For this reason, it is recommended to avoid ``std::vector<bool>`` for anything but a read-only
-input with the C++ standard execution policies.
+.. note::
+   Implementations of ``std::vector<bool>`` are not required to avoid data races for concurrent modifications
+   of vector elements. Some implementations may optimize multiple ``bool`` elements into a bitfield, making it unsafe
+   for multithreading. For this reason, it is recommended to avoid ``std::vector<bool>`` for anything but a read-only
+   input with the C++ standard aligned execution policies.
 
 When using a device execution policy, you can use one of the following ways to pass data to an algorithm:
 
-* ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` functions
-* Unified shared memory (USM) pointers
-* ``std::vector`` with or without a USM allocator
+* ``oneapi::dpl::begin`` and ``oneapi::dpl::end`` functions for a `SYCL buffer
+  <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:buffers>`_;
+* `Unified shared memory (USM) <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:usm>`_ pointers;
+* ``std::vector`` with or without a USM allocator.
+
+For an algorithm to access data, it is important that the used execution policy matches the data storage type.
+The following table shows which execution policies can be used with various data storage types.
+
+====================================== =============== =============
+Data Storage                           Device Policies Host Policies
+====================================== =============== =============
+SYCL buffer                            Yes             No
+Device-allocated USM                   Yes             No
+Shared and host-allocated USM          Yes             Yes
+``std::vector`` with a USM allocator   Yes             Yes
```

Comment:
I think should be like that:
```std::vector`` USM allocator with a shared or host types`

Reply:
There is no such thing as a USM allocator for USM device memory (intentional choice in SYCL specification).  It still could be useful to point it out for clarity, but I believe this is technically true as written.  (Again, we want to encourage people to use the direct pointers to the underlying data if they used this type, rather than the iterators.)

Reply:
Dan is correct, see https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_c_allocator_interface